docs/guides/file-system-based-automated-bundle-generation.md

```diff
@@ -0,0 +1,133 @@
+```
+
+Now, To automatically register, `A_ComponentOne`, `B_ComponentOne`, `B_ComponentTwo` for the usage with `react_component` and `react_component_hash` helpers, Create directory structure as mentioned below:
+
+```
+app/javascript:
+  └── packs:                   
+  │   └── application.js     
+  │   └── application.css
+  └── src:                   
+  │   └── A
+  │   │ └── ...
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   └── A_ComponentOne.jsx
+  │   └── B
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   │ └── B_ComponentOne.jsx
+  │   │   │ └── B_ComponentTwo.jsx
+public/packs                
+```
+
+To register react component, It is no longer needed to create a pack entry and manually register it by calling `ReactOnRails.register`. With automatically generated packs, you can directly use `A_ComponentOne`, `B_ComponentOne` and `B_ComponentTwo` in rails view using:
+
+```erb
+    <%= react_component("A_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentTwo", {}, auto_load_bundle: true) %>    
+```
+
+If `A_ComponentOne` uses multiple HTML strings for server rendering, [`react_component_hash`](https://www.shakacode.com/react-on-rails/docs/api/view-helpers-api/#react_component_hash) view helper can be used on rails view as illustrated below.
+
+```erb
+<% A_ComponentOne = react_component_hash("A_ComponentOne", 
+                                             prerender: true,
+                                             auto_load_bundle: true
+                                             props: {}
+                                          ) %>   
+<% content_for :title do %>
+   <%= A_ComponentOne['title'] %>
+<% end %>
+<%= A_ComponentOne["componentHtml"] %>
+```
+
+the default value of the auto_load_bundle parameter can be specified by setting `config.auto_load_bundle` in `config/initializers/react_on_rails.rb`.
+
+### Server Rendering and Client Rendering Components
+
+If server rendering is enabled, the component will be registered for the usage both on server and client rendering. In order to have separate definitions for client and server rendering, Name the components files as `Component_Name.server.jsx` and `Component_Name.client.jsx`. During the server rendering, the `Component_Name.server.jsx` definition will be used, and for client rendering, `Component_Name.client.jsx` file will be used.
+
+### Using Automated Bundle Generation Feature with already defined packs
+
+To use Automated Bundle Generation Feature with already defined packs, `config/initializers/react_on_rails` should explicitly be configured with `config.auto_load_bundle = false`, in order to avoid automated loading of component packs. For the components using Bundle Generation Feature, explicitly pass `auto_load_bundle` option in `react_component` and `react_component_hash`.
```

Comment:
"should explicitly be configured with `config.auto_load_bundle = false`" or you can just add it to the calls, right?